modules/examples/todolist-service/src/main/scala/todo/service/TagService.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2017-2018 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package todo.service
+
+import cats.Monad
+import cats.implicits._
+import freestyle.tagless._
+import freestyle.tagless.logging.LoggingM
+import todo.model.Tag
+import todo.persistence.TagRepository
+
+@module
+trait TagService[F[_]] {
+
+  implicit val M: Monad[F]
+
+  val repo: TagRepository[F]
+  val log: LoggingM
+
+  private val model: String = classOf[Tag].getSimpleName
+
+  def insert(item: Tag): F[Option[Tag]] =
+    for {
+      _            <- log.debug(s"Trying to insert a ${model}")
```

Comment:
`$model`?

Reply:
I know that `${...}` could be replace with `$...` if it is not accessing attributes or methods inside the object. It just my preference. But if it is really necessary, I could change them.